Sources/libhostmgr/Foundation+Extensions.swift

```diff
@@ -25,19 +48,53 @@ extension FileManager {
         return fileExists(atPath: url.path, isDirectory: &isDir) && isDir.boolValue
     }
 
+    public func availableStorageSpace(forVolumeContainingDirectoryAt url: URL) throws -> Int64 {
+        let values = try url.resourceValues(forKeys: [.volumeAvailableCapacityForImportantUsageKey])
+        return values.volumeAvailableCapacityForImportantUsage ?? 0
+    }
+
+    public func size(ofObjectAt url: URL) throws -> Int {
+
+        var isDir: ObjCBool = true
+        guard fileExists(atPath: url.path, isDirectory: &isDir) else {
+            return 0
+        }
+
+        if isDir.boolValue {
+            return try directorySize(of: url)
+        } else {
+            return try fileSize(of: url)
+        }
+    }
+
+    func fileSize(of url: URL) throws -> Int {
+        let values = try url.resourceValues(forKeys: [.fileSizeKey])
+        return values.fileSize!
+    }
+
+    /// returns total allocated size of a the directory including its subFolders or not
+    func directorySize(of url: URL) throws -> Int {
+        guard
+            let enumerator = enumerator(at: url, includingPropertiesForKeys: [.totalFileAllocatedSizeKey]),
+            let urls = enumerator.allObjects as? [URL]
+        else {
+            return 0
+        }
+
+        let sizes = try urls
+            .compactMap { try $0.resourceValues(forKeys: [.totalFileAllocatedSizeKey]).totalFileAllocatedSize }
+
+        return sizes.reduce(0, +)
+     }
+
     public func createTemporaryFile(containing string: String = "") throws -> URL {
         let path = temporaryFilePath()
         try string.write(to: path, atomically: false, encoding: .utf8)
         return path
     }
 
-    public func temporaryFilePath() -> URL {
-        self.temporaryDirectory
-            .appendingPathComponent(UUID().uuidString + ".tmp")
-    }
-
-    public func createDirectoryTree(atUrl url: URL) throws {
```

Comment:
Removed this one because it's a very thin wrapper around an existing API and doesn't add much value